shell/browser/api/electron_api_session.cc

```diff
@@ -150,6 +154,42 @@ uint32_t GetQuotaMask(const std::vector<std::string>& quota_types) {
   return quota_mask;
 }
 
+constexpr content::BrowsingDataRemover::DataType kClearDataTypeAll =
+    0xffffffffffffffffull;
+constexpr content::BrowsingDataRemover::OriginType kClearOriginTypeAll =
+    content::BrowsingDataRemover::ORIGIN_TYPE_UNPROTECTED_WEB |
+    content::BrowsingDataRemover::ORIGIN_TYPE_PROTECTED_WEB;
+
+// Observes the BrowsingDataRemover that backs the `clearBrowsingData` method
+// and resolves/rejects that API's promise once its done
+class ClearBrowsingDataObserver
+    : public content::BrowsingDataRemover::Observer {
+ public:
+  ClearBrowsingDataObserver(gin_helper::Promise<void> promise,
+                            content::BrowsingDataRemover* remover)
+      : promise_(std::move(promise)) {
+    observation_.Observe(remover);
+  }
+
+  void OnBrowsingDataRemoverDone(
+      content::BrowsingDataRemover::DataType failed_data_types) override {
+    if (failed_data_types == 0) {
+      promise_.Resolve();
+    } else {
+      promise_.RejectWithErrorMessage(absl::StrFormat(
```

Comment:
`absl::StrFormat()` [is currently banned](https://chromium.googlesource.com/chromium/src/+/main/styleguide/c++/c++-features.md#string-formatting-banned); use `base::StringPrintf()` for now instead.

```suggestion
      promise_.RejectWithErrorMessage(base::StringPrintf(
```

Reply:
Ah, I had seen the bug tracking migration to `absl::StrFormat` but did not see that it's banned until an official switch over. Good catch!